components/event-service/internal/externalapi/genericeventhandler.go

```diff
@@ -0,0 +1,149 @@
+package externalapi
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/kyma-project/kyma/components/event-service/internal/events/api"
+	apiv1 "github.com/kyma-project/kyma/components/event-service/internal/events/api/v1"
+	"github.com/kyma-project/kyma/components/event-service/internal/events/mesh"
+	"github.com/kyma-project/kyma/components/event-service/internal/events/shared"
+	"github.com/kyma-project/kyma/components/event-service/internal/httpconsts"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	requestBodyTooLargeErrorMessage = "http: request body too large"
+)
+
+var (
+	isValidEventTypeVersion = regexp.MustCompile(shared.AllowedEventTypeVersionChars).MatchString
+	isValidEventID          = regexp.MustCompile(shared.AllowedEventIDChars).MatchString
+	traceHeaderKeys         = []string{"x-request-id", "x-b3-traceid", "x-b3-spanid", "x-b3-parentspanid", "x-b3-sampled", "x-b3-flags", "x-ot-span-context"}
+)
+
+type maxBytesHandler struct {
+	next  http.Handler
+	limit int64
+}
+
+func (h *maxBytesHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	r.Body = http.MaxBytesReader(rw, r.Body, h.limit)
+	h.next.ServeHTTP(rw, r)
+}
+
+// NewEventsHandler creates an http.Handler to handle the events endpoint
+func NewEventsHandler(config *mesh.Configuration, maxRequestSize int64) http.Handler {
+	return &maxBytesHandler{next: http.HandlerFunc(getEventsHandler(config)), limit: maxRequestSize}
+}
+
+type permanentRedirectionHandler struct {
+	location string
+	handler  http.Handler
+}
+
+func (h *permanentRedirectionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Location", h.location)
+	w.WriteHeader(http.StatusMovedPermanently)
+}
+
+// NewPermanentRedirectionHandler creates an http.Handler to handle the /v2/events legacy endpoint
+func NewPermanentRedirectionHandler(redirectLocation string) http.Handler {
+	return &permanentRedirectionHandler{location: redirectLocation}
+}
+
+func getEventsHandler(config *mesh.Configuration) func(w http.ResponseWriter, req *http.Request) {
+	return func(w http.ResponseWriter, req *http.Request) {
+		// parse request body to PublishRequestV1
+		if req.Body == nil || req.ContentLength == 0 {
+			resp := shared.ErrorResponseBadRequest(shared.ErrorMessageBadPayload)
+			writeJSONResponse(w, resp)
+			return
+		}
+
+		var err error
+		parameters := &apiv1.PublishEventParametersV1{}
+		decoder := json.NewDecoder(req.Body)
+		err = decoder.Decode(&parameters.PublishrequestV1)
+		if err != nil {
+			var resp *api.PublishEventResponses
+			if err.Error() == requestBodyTooLargeErrorMessage {
+				resp = shared.ErrorResponseRequestBodyTooLarge(err.Error())
```

Comment:
It would be nice if the response could somehow include the maximum size in the message.

Reply:
Keeping the message, to respect the previous contract